tests/integration.rs

```diff
@@ -341,3 +372,89 @@ fn subcommand_usage() -> TestResult {
 fn standalone_usage() -> TestResult {
     golden_reference(&["-h"], "tests/standalone-usage.txt")
 }
+
+/// This test is a bit more verbose than the other tests because it uses a slightly
+/// different mechanism for setting up and sweeping / cleaning.
+///
+/// Step-by-step:
+///
+/// * Create a new temporary directory. Other tests only output the built files into
+///   the target directory, but this test copies the whole template project into the
+///   target directory.
+/// * Build the two different (nested) projects using cargo build, but without incremental
+///   compilation, because cargo-sweep doesn't remove files built by incremental compilation,
+///   which affects this test since this test doesn't use a separate target folder for just
+///   the target files.
+/// * Do a dry run sweep, asserting that nothing was actually cleaned.
+/// * Do a proper sweep, asserting that the target directory size is back down to its
+///   expected size.
+///
+/// Please note that there is some ceremony involved, namely correctly setting:
+/// * `CARGO_TARGET_DIR` to `unset`, so as not to clash with users who use `CARGO_TARGET_DIR`
+///   when they invoke `cargo test` for running these tests.
+/// * `CARGO_INCREMENTAL` to `0`, because cargo-sweep doesn't yet clear incremental files.
+#[test]
+fn recursive_multiple_root_workspaces() -> TestResult {
+    let target = tempdir()?;
+
+    let nested_workspace_dir = test_dir().join("nested-root-workspace");
+    let options = CopyOptions::default();
+
+    // Copy the whole nested-root-workspace folder (and its content) into the new temp dir,
+    // and then `cargo build` and run the sweep tests inside that directory.
+    fs_extra::copy_items(&[&nested_workspace_dir], target.path(), &options)?;
+
+    let old_size = get_size(target.path())?;
+
+    // Build bin-crate
+    run(Command::new(env!("CARGO"))
+        // If someone has built & run these tests with CARGO_TARGET_DIR,
+        // we need to override that.
+        .env_remove("CARGO_TARGET_DIR")
+        // Don't output incremental build artifacts
+        .env("CARGO_INCREMENTAL", "0")
+        .arg("build")
+        .current_dir(target.path().join("nested-root-workspace/bin-crate")));
+
+    let intermediate_build_size = get_size(target.path())?;
+    assert!(intermediate_build_size > old_size);
+
+    // Build workspace crates
+    run(Command::new(env!("CARGO"))
+        // If someone has built & run these tests with CARGO_TARGET_DIR,
+        // we need to override that.
+        .env_remove("CARGO_TARGET_DIR")
+        // Don't output incremental build artifacts
+        .env("CARGO_INCREMENTAL", "0")
+        .arg("build")
+        .current_dir(target.path().join("nested-root-workspace")));
```

Comment:
nit: Please deduplicate these `cargo` commands - you can change the existing `cargo` command to require an absolute path instead of a relative path as the `project` argument.

Reply:
Done